maro/simulator/scenarios/finance/business_engine.py

```diff
@@ -0,0 +1,653 @@
+            ):
+                executing_volume = self.adjust_trade(
+                    action.direction, executing_price, executing_volume,
+                    executing_commission, executing_tax, remaining_money
+                )
+                adjusted = True
+                # print(f"Adjust volume for trade:{executing_volume}")
+                continue
+
+        if executing_volume > 0:
+            ret = TradeResult(executing_volume, executing_price, executing_commission, executing_tax)
+            rets.append(ret)
+        else:
+            pass
+
+        return rets
+
+    def validate_constrant(
+        self, odd_shares: int, direction: OrderDirection, executing_price: float, executing_volume: int
+    ):
+        ret = True
+        if direction == OrderDirection.buy:
+            if executing_volume % self._conf["trade_constraint"]["min_buy_unit"] != 0:
+                ret = False
+        if direction == OrderDirection.sell:
+            odd_volume = executing_volume % self._conf["trade_constraint"]["min_sell_unit"]
+            if odd_volume != 0:
+                if odd_shares != odd_volume:
+                    ret = False
+
+        return ret
+
+    def adjust_constraint(
+        self, odd_shares: int, direction: OrderDirection, executing_price: float, executing_volume: int
+    ):
+        ret = executing_volume
+        if direction == OrderDirection.buy:
+            odd_volume = ret % self._conf["trade_constraint"]["min_buy_unit"]
+            ret -= odd_volume
+        if direction == OrderDirection.sell:
+            odd_volume = executing_volume % self._conf["trade_constraint"]["min_sell_unit"]
+            if odd_volume != 0:
+                ret -= odd_volume
+                ret += odd_shares
+
+        return int(ret)
+
+    def validate_trade(
+        self, direction: OrderDirection, executing_price: float,
+        executing_volume: int, executing_commission: float, executing_tax: float, remaining_money: float
+    ):
+        ret = True
+        if direction == OrderDirection.buy:
+            money_needed = executing_price * executing_volume + executing_commission + executing_tax
+            if money_needed > remaining_money:
+                ret = False
+        else:
+            money_needed = executing_commission + executing_tax
+            if money_needed > remaining_money + executing_price * executing_volume:
+                ret = False
+
+        return ret
+
+    def adjust_trade(
+        self, direction: OrderDirection, executing_price: float,
+        executing_volume: int, executing_commission: float, executing_tax: float, remaining_money: float
+    ):
+        ret = executing_volume
+        if direction == OrderDirection.buy:
+            money_needed = executing_price * executing_volume + executing_commission + executing_tax
+            ret = ret - math.ceil(
+                (money_needed - remaining_money) / (executing_price * self._conf["trade_constraint"]["min_buy_unit"])
```

Comment:
should we initialize 
self._min_buy_unit = self._conf["trade_constraint"]["min_buy_unit"]
in self._init()?

or using other meaning name